linkedin/parseq
# Migrate to Gradle build automation

Changes:
- Change the hierarchy structure;
  - Set the all the contrib project as ParSeq's subproject.
  - Set the core source code of Parseq as a subproject(module)
  - Rename the previous fmpp module and set it as a code generation module in parseq module
  - Rename the preivous src-test module and set it as test utility module
  - Move previous test module as a part of parseq subproject
- For each contrib project, convert the Maven configuration to Gradle, including set dependencies, package source jars, javadoc jars and build fatjar in  project like parseq-tracevis-server, parseq-lambda-names.
- For the root project, set all the shared dependencies and general configuration
- For the front tool project, migrate its makefile to Gradle, and wrap the npm commands in plugin.
- Support the function to sign up the .jar files,  and to publish to Maven Central Repository.

How to test:
- Set up your GPG key and password in the "gradle.properties" file.
- Run "gradle build" to build the jar files by Gradle and Run "ant clean build test dist bundle" to build jar files by Ant.
- Compare the contents of jar files between Gradle and Ant to check the difference.
- For the parseq-tracevis-server-<version>-jar-with-dependencies.jar, Use "java -jar parseq-tracevis-server-<version>-jar-with-dependencies.jar /usr/bin/dot" to see if you can start a server in localhost.
- For the transitive dependencies for both build automation, you can write a script to fetch all the dependencies and compare them.

Note:
- For MANIFEST files, they could not be exactly the same to the MANIFEST files in Maven.
- For Javadoc files, some of the pages' "Tree" tab point to "overview-tree.html" but not "package-tree.html", it may a little different to Maven.
- For project "parseq-examples" and "parseq-legacy-examples", we're not going to push their jar to Maven repo.
